src/coreclr/vm/jithelpers.cpp

```diff
@@ -629,48 +672,68 @@ HCIMPL1_V(UINT64, JIT_Dbl2ULngOvf, double val)
 #endif // _DEBUG
         return ret;
     }
-
     FCThrow(kOverflowException);
+#endif //TARGET_X86 || TARGET_AMD64
 }
 HCIMPLEND
 
-
-#if !defined(TARGET_X86) || defined(TARGET_UNIX)
-
-HCIMPL1_V(INT64, JIT_Dbl2Lng, double val)
+HCIMPL1_V(UINT32, JIT_Dbl2UInt, double val)
 {
     FCALL_CONTRACT;
+#if defined(TARGET_X86) || defined(TARGET_AMD64)
+    // The code below works to saturate the input value
+    //      val = NaN or val <= 0; output = 0
+    //      0 < val < UINT32_MAX;  output = (uint32_t)val
+    //      UINT32_MAX <= val;     output = UINT32_MAX
+    const double uint_max = 4294967295.0;
+    return (val > 0) ? ((val >= uint_max) ? UINT32_MAX : (UINT32)val) : 0;
```

Comment:
```suggestion
    return (val >= 0) ? ((val >= uint_max) ? UINT32_MAX : (UINT32)val) : 0;
```
Something to consider here and in other similar places. It would avoid mis-predicted branches when this method is called on a random mix of zeros and non-overflowing numbers that is likely a common scenario.